python/pyspark/sql/types.py

```diff
@@ -189,7 +189,7 @@ def toInternal(self, dt):
         if dt is not None:
             seconds = (calendar.timegm(dt.utctimetuple()) if dt.tzinfo
```

Comment:
I just tested it. In Python3, even `toInternal` returns Python's `long`, you still can a `java.lang.Integer` in JVM side.

However, in Python2, you can get `java.lang.Long`.